src/main/java/uk/nhs/hee/tis/trainee/credentials/dto/PlacementCredentialDto.java

```diff
@@ -117,4 +118,17 @@ public String getTisId() {
   public UUID getUniqueIdentifier() {
     return uniqueIdentifier;
   }
+
+  @JsonIgnore
+  public int calculatePlacementHashCode() {
+    int result = 0;
+    result = result + (specialty.hashCode());
+    result = result + (grade.hashCode());
+    result = result + (nationalPostNumber.hashCode());
+    result = result + (employingBody.hashCode());
+    result = result + (site.hashCode());
+    result = result + (startDate.hashCode());
+    result = result + (endDate.hashCode());
+    return result;
```

Comment:
This has the same issues as the previous iteration, the built in hashcode functions are not guaranteed to be consistent between Java version, or even deployments/invocations.

Take another look at the md5 hashing example I previously sent.